modules/dnn/src/ie_ngraph.hpp

```diff
@@ -91,16 +91,18 @@ class InfEngineNgraphNode : public BackendNode
 {
 public:
     InfEngineNgraphNode(const std::vector<Ptr<BackendNode> >& nodes, Ptr<Layer>& layer,
-                        std::vector<Mat*>& inputs, std::vector<Mat>& outputs,
+                        std::vector<Mat*>& inputs, std::vector<Mat>& outMats,
                         std::vector<Mat>& internals);
 
     InfEngineNgraphNode(std::shared_ptr<ngraph::Node>&& _node);
     InfEngineNgraphNode(std::shared_ptr<ngraph::Node>& _node);
 
     void setName(const std::string& name);
+    ngraph::Output<ngraph::Node> GetOidOutput() const;
 
     // Inference Engine network object that allows to obtain the outputs of this layer.
     std::shared_ptr<ngraph::Node> node;
```

Comment:
Maybe we just can replace `node` type to `ngraph::Output<ngraph::Node>`?

Reply:
We need to store `ngraph::OutputVector` and `oid`. And we have some logic with nodes to release them after creating ngraph_function.

Reply:
Which logic? Can we use fields of ngraph::Output instead? Can we use Node* from ngraph::Output where it's not possible?
My main goal is to reduce number of changed files. Each layer returns ngraph::Output and std::shapred_ptr<Node> can be casted to each other explicitly (i.e. each layer accepts ngraph::Output as input but we pass shared ptr).

Reply:
Each layer returns `ngraph::OutputVector`. And we need to get specific` ngraph::Output` inside layer. Logic with setting names for node. If store OutputVector, then we have problems with same names.

Reply:
But not to drop `std::shared_ptr<Node>`?

Reply:
To store node name

Reply:
So maybe we can use `get_node()`  to get Node* if needed?

https://github.com/NervanaSystems/ngraph/blob/0af332260d9dd44a5722a1664f48781cef4c6456/src/ngraph/node_output.hpp#L73

Reply:
How it helps to reduce number of changed files? I think if we store `ngraph::OutputVector` instead of `std::shared_ptr<Node>` this increases number of changed lines. Used get_node() is less convenient than a node directly.

Reply:
In example, can we keep
```patch
-    auto& ieInpNode = nodes[0].dynamicCast<InfEngineNgraphNode>()->node;
+    auto ieInpNode = nodes[0].dynamicCast<InfEngineNgraphNode>()->GetOidOutput();
```
is `node` will be `ngraph::Output<ngraph::Node>`?

```
ngraph::Output<ngraph::Node> InfEngineNgraphNode::GetOidOutput() const {
    return node->output(oid);
}
```

Reply:
1. We need to store ngraph::OutputVector.
2. Inside layers we need get current ngraph::Output<ngraph::Node>